# Example how to produce MDavocado diagrams

This Jupyter notebook will giude you through a simple procedure how to produce MDavocado diagrams from one example molecular dynamics (MD) simulation trajectory. In order to be able to run this notebook you need to be in your MDavocado conda environment as explained in more detail [here](https://github.com/zoranstefanic/MDavocado)

Basically we need to install all the required packages by running:

```bash
conda create --name MDavocado --file packages.txt
conda activate MDavocado
```

Now you have to go to the **examples** subdirectory and run:

```
jupyter notebook Example.ipynb
```

It will open this notebook in your browser and you can ineractivelly run it.

In [1]:
from MDAnalysisData import datasets

This will import all the necessary stuff to be able to fetch the example trajectories included in the [MDAnalysisData](https://www.mdanalysis.org/MDAnalysisData/index.html) package. As a first example we will produce the MDavocado diagrams for the [AdK equilibrium dataset](https://www.mdanalysis.org/MDAnalysisData/adk_equilibrium.html).

In [2]:
# Fetch the AdK dataset
adk = datasets.fetch_adk_equilibrium()

Here we can see the basic info about this trajectory (it is only 161 MB!):

In [5]:
print(adk['DESCR'])

.. -*- coding: utf-8 -*-

.. _`adk-equilibrium-dataset`:

AdK equilibrium trajectory dataset

MD trajectory of apo adenylate kinase with CHARMM27 force field and
simulated with explicit water and ions in NPT at 300 K and 1
bar. Saved every 240 ps for a total of 1.004 µs. Produced on PSC
Anton. The trajectory only contains the protein while all solvent was
stripped in order to reduce the trajectory size. The whole protein was
superimposed on the CORE domain of AdK by RMSD fitting.

The topology is contained in the PSF file (CHARMM format). The
trajectory is contained in the DCD file (CHARMM/NAMD format).


Notes
-----

Data set characteristics:

 :size: 161 MB
 :number of trajectories: 1
 :number of frames:  4187
 :number of particles: 3341
 :creator: Sean Seyler
 :URL:  `10.6084/m9.figshare.5108170.v1 <https://doi.org/10.6084/m9.figshare.5108170.v1>`_
 :license: `CC-BY 4.0 <https://creativecommons.org/licenses/by/4.0/legalcode>`_
 :reference: [Seyler2017]_
    

.. [Seyler2017]  Seyler

We make sure we are in the top MDavocado directory:

In [7]:
cd ..

/disk1/ALOKOMP/MDavocado


In [8]:
from ramachandran import *

In [10]:
# We create the MDanalysis Universe from the above trajectory and topology
u = Universe(adk['topology'],adk['trajectory'])

# We initialize the main RamachandranPlots class
rp = RamachandranPlots(u)

# And simply use the run method of the above class
rp.run()

/home/zoran/anaconda3/envs/MDavocado/lib/python3.12/site-packages/MDAnalysis/coordinates/DCD.py:165: DeprecationWarning: DCDReader currently makes independent timesteps by copying self.ts while other readers update self.ts inplace. This behavior will be changed in 3.0 to be the same as other readers. Read more at https://github.com/MDAnalysis/mdanalysis/issues/3889 to learn if this change in behavior might affect you.
  warnings.warn("DCDReader currently makes independent timesteps"
/home/zoran/anaconda3/envs/MDavocado/lib/python3.12/site-packages/MDAnalysis/analysis/dihedrals.py:400: UserWarning: Cannot determine phi and psi angles for the first or last residues
  warnings.warn("Cannot determine phi and psi angles for the first "


  0%|          | 0/418 [00:00<?, ?it/s]

Writing out rama_0_418.pkl


  0%|          | 0/418 [00:00<?, ?it/s]

Writing out rama_418_836.pkl


  0%|          | 0/418 [00:00<?, ?it/s]

Writing out rama_836_1254.pkl


  0%|          | 0/418 [00:00<?, ?it/s]

Writing out rama_1254_1672.pkl


  0%|          | 0/418 [00:00<?, ?it/s]

Writing out rama_1672_2090.pkl


  0%|          | 0/418 [00:00<?, ?it/s]

Writing out rama_2090_2508.pkl


  0%|          | 0/418 [00:00<?, ?it/s]

Writing out rama_2508_2926.pkl


  0%|          | 0/418 [00:00<?, ?it/s]

Writing out rama_2926_3344.pkl


  0%|          | 0/418 [00:00<?, ?it/s]

Writing out rama_3344_3762.pkl


  0%|          | 0/425 [00:00<?, ?it/s]

Writing out rama_3762_4187.pkl
df0
Finished directory df0
df1
Finished directory df1
df2
Finished directory df2
df3
Finished directory df3
df4
Finished directory df4
df5
Finished directory df5
df6
Finished directory df6
df7
Finished directory df7
df8
Finished directory df8
df9
Finished directory df9
Copying blanks to df0
Copying blanks to df1
Copying blanks to df2
Copying blanks to df3
Copying blanks to df4
Copying blanks to df5
Copying blanks to df6
Copying blanks to df7
Copying blanks to df8
Copying blanks to df9
Adding labels in dir: df0
Adding labels in dir: df1
Adding labels in dir: df2
Adding labels in dir: df3
Adding labels in dir: df4
Adding labels in dir: df5
Adding labels in dir: df6
Adding labels in dir: df7
Adding labels in dir: df8
Adding labels in dir: df9
Making gif: 1
Making gif: 2
Making gif: 3
Making gif: 4
Making gif: 5
Making gif: 6
Making gif: 7
Making gif: 8
Making gif: 9
Making gif: 10
Making gif: 11
Making gif: 12
Making gif: 13
Making gif: 14
Making gif: 15
Mak

After a few minutes (it could be up to few hours for large/long trajectories) you get a "All done!" message.

This means that you should have a MDavocado diagram in form of a GIF image produced in the directory (usually it is set to ~/MDAnalysis_data/adk_equilibrium/')

For this small example trajectory it will look like this:

In [14]:
cd ~/ALOKOMP/MDavocado/

/disk1/ALOKOMP/MDavocado


![MDavocado diagram of AdK example trajectory](./A.gif)

This is a rather short trajectory with just 4187 frames and therefore there are not much points to aggregate, although the dynamics of several moving amino acids is still very visible. But in case of much longer trajectories (as in the case of trajectory having 400 000 frames and lasting 1 microsecond from [ALOKOMP project](https://alokomp.irb.hr/) page) you would get something like this:

![](https://alokomp.irb.hr/static/avokado/1458/A.gif)